fieldservice_maintenance/models/maintenance_request.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (C) 2020 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class MaintenanceRequest(models.Model):
+    _inherit = "maintenance.request"
+
+    fsm_order_id = fields.Many2one("fsm.order", "Field Service Order")
```

Comment:
Why don't use a one2many fields with inverse_name = request_id ?
With your current implementation when you create the maintenance.request from the fsm.order your fsm_order_id is empty. 
You should have a same result from both side therefore it is better to use a one2many.

If you need a "one2one" relation I think it is better to apply a constrains()